docker/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM python:3.7-slim
+
+RUN apt-get update -q \
+    && apt-get install --no-install-recommends -y \
+        ffmpeg \
+        gcc \
+        git \
+        libcairo2-dev \
+        libffi-dev \
+        libpango1.0-dev \
+        pkg-config \
+        wget
+
+# setup a minimal texlive installation
+COPY ./texlive-profile.txt /tmp/
+ENV PATH=/usr/local/texlive/bin/x86_64-linux:$PATH
+RUN wget -O /tmp/install-tl-unx.tar.gz http://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz && \
+    mkdir /tmp/install-tl && \
+    tar -xzf /tmp/install-tl-unx.tar.gz -C /tmp/install-tl --strip-components=1 && \
+    /tmp/install-tl/install-tl --profile=/tmp/texlive-profile.txt \
+    && tlmgr install \
+        amsmath babel-english cm-super doublestroke dvisvgm fundus-calligra \
+        jknapltx latex-bin microtype ms physics preview ragged2e relsize rsfs \
+        setspace standalone tipa wasy wasysym xcolor xkeyval
+
+# clone and build manim
+RUN git clone --depth 1 --branch v0.1.0 https://github.com/ManimCommunity/manim.git /opt/manim
+WORKDIR /opt/manim
+RUN pip install --no-cache .
+
+# ensure ffi bindings are generated correctly
+WORKDIR /usr/local/lib/python3.7/site-packages
+RUN python cairocffi/ffi_build.py \
+    && python pangocffi/ffi_build.py \
+    && python pangocairocffi/ffi_build.py
+
+# create working directory for user to mount local directory into
+WORKDIR /manim
+RUN chmod 666 /manim
```

Comment:
Can you explain to me what this does?

Reply:
The intended way of how to use this image is to mount the local working directory into the `/manim` directory in the container (see https://hub.docker.com/r/manimcommunity/manim). When doing so, you usually want to pass `--user="$(id -u):$(id -g)"` to the docker command so that the user within the container uses the same UID and GID as outside: this is important so that files created within the container are owned by the user outside of the container. (Otherwise they are owned by the user with uid/gid 1000, which is usually root and thus creates files that need admin privileges to delete, which is annoying.)

Without changing the permissions of `/manim` to `666` (read and write permissions for everyone), the manim process in the container is not allowed to create the corresponding files.